dataprofiler/data_readers/data_utils.py

```diff
@@ -180,7 +204,7 @@ def read_json_df(data_generator, selected_columns=None, read_in_string=False):
                 ),
                 ignore_dicts=True,
             )
-            lines.append(obj)
+            lines.append(cast(Dict, obj))
```

Comment:
did we try overloading `unicode_to_str`? would that fix this so we don't have to cast?

Reply:
I did some experimentation and added this:
```py
@overload
def unicode_to_str(data: Dict, ignore_dicts: bool) -> Dict: ...
@overload
def unicode_to_str(data: str, ignore_dicts: bool) -> str: ...
@overload
def unicode_to_str(data: List, ignore_dicts: bool) -> List: ...
def unicode_to_str(
    data: Union[str, List, Dict], ignore_dicts: bool = False
) -> Union[str, List, Dict]:
# function body here
```
The issue I'm getting now is when we call this
```py
json.loads(raw_line, object_hook=unicode_to_str, object_pairs_hook=OrderedDict)
```
I get 
```
Argument "object_hook" to "loads" has incompatible type overloaded function; expected "Optional[Callable[[Dict[Any, Any]], Any]]"
```
from mypy. My guess is that we can solve this by casting before we pass in the function to `object_hook` but that would somewhat defeat the purpose of overloading in the first place.

Reply:
@Sanketh7 I think you don't need to specify:
```
def unicode_to_str(
    data: Union[str, List, Dict], ignore_dicts: bool = False
) -> Union[str, List, Dict]:
```
b/c of the three other overloads. I assume the defaults would still need to be assigned though

Reply:
Even with that removal, I'm still getting the same issue. I think it has to do with the function taking on 3 possible types while `object_hook` is only compatible with one of those types.

Reply:
This issue no longer applies due to the new way of handling json types.